android-client-sdk/src/main/java/com/devcycle/sdk/android/api/EventQueue.kt

```diff
@@ -99,8 +103,16 @@ internal class EventQueue constructor(
             }
         }
 
-        if (eventQueue.size > 0 || !result.success) {
-            scheduler.scheduleWithDelay { run() }
+        if ((eventQueue.size > 0 || !result.success) && !isClosed.get()) {
+            scheduleJob = scheduler.scheduleWithDelay { run() }
+        }
+
+        // call the provided callback function if the queue is now empty, or if there was an error
+        if (eventQueue.size == 0 || !result.success) {
+            if (result.exception != null) {
```

Comment:
if "success" is already false here (or there are still events in the queue) should we just call onError no matter what? We can pass the exception if it exists, but could have a "fallback" generic exception as well rather than checking its existence here

Reply:
no, because success is false in non-error cases too, such as when the queue is empty

Reply:
I think it also stays false in the case where we dont have access to the lock because the events are already flushing

Reply:
or do you think that not having the lock would warrant throwing an error?

Reply:
not having the lock is fine, though that does present another problem I hadn't thought of for this close callback..it almost would need to wait for the lock to be available

Reply:
yeah, thats a bit of a problem